internal/testutils/xds/e2e/setup_management_server.go

```diff
@@ -110,3 +111,30 @@ func DefaultBootstrapContents(nodeID, serverURI string) ([]byte, error) {
 	}
 	return bs, nil
 }
+
+// SetupDNS unregisters the DNS resolver and registers a manual resolver for the
+// same scheme. This allows the test to mock the DNS resolution by supplying the
+// addresses of the test backends.
+//
+// Returns the following:
+//   - a channel onto which the DNS target being resolved is written to by the
+//     mock DNS resolver
+//   - a channel to notify close of the DNS resolver
+//   - a channel to notify re-resolution requests to the DNS resolver
+//   - a manual resolver which is used to mock the actual DNS resolution
+//   - a cleanup function which re-registers the original DNS resolver
+func SetupDNS() (chan resolver.Target, chan struct{}, chan resolver.ResolveNowOptions, *manual.Resolver, func()) {
```

Comment:
Done. I am questioning the ergonomy of this function though. Would it be enough to just return the manual resolver, for most test cases?
Asserting for the exact names that were resolves sounds like it would be useful to test the exact interaction between the channel and the resolver, but I wonder how generally useful that is, and perhaps it encourages brittle tests?

Reply:
Yes, we have been having discussions about keeping these setup-y kind of test functions, to be as simple as possible, and keep setup code close to test logic, so that it makes reading the test easier.

Yeah, I do agree that this function does more than would be required for most tests. Some tests use the returned channels (one or all of the returned ones), while some tests don't care about the returned channels, and only want the manual resolver.

I would be fine at this point with whatever approach you want to take:
- Completely get rid of this, and inline the mocking part in each of your tests. Even though that would be duplicate code, that seems to be the recommended approach for tests, instead of having complicated setup code that lives outside of the test logic
- Leave it as it and maybe add a TODO to perform cleanup later.

Reply:
I did option 1: duplicate DNS setup in tests that need it, and removed unused return values.